### PR TITLE
TOK-365: claim rewards by asset

### PR DESCRIPTION
**Pull request by franciscotobar**
## What

- Claim all rewards by asset

## Why

- There was no option to claim rewards by individual asset

## Testing

- Unit tests have been updated & added, and all previous flows should work as before.

## Refs

- [TOK-365](https://rsklabs.atlassian.net/browse/TOK-365)

### CODE REVIEW

File: src/SponsorsManager.sol [hunk 1]
```diff
@@ -231,6 +231,20 @@ contract SponsorsManager is BuilderRegistry {
         }
     }
 
+    /**
+     * @notice claims sponsor rewards from a batch of gauges
+     * @param gauges_ array of gauges to claim
+     * @param rewardToken_ address of the token rewarded
+     *  address(uint160(uint256(keccak256("COINBASE_ADDRESS")))) is used for coinbase address
+     */
+    function claimSponsorRewards(address rewardToken_, Gauge[] memory gauges_) external {
```
Comment (franciscotobar) [inline src/SponsorsManager.sol:240]: Im not sure if we want to add something like we did with the incentivize:

- [rewardToken](https://github.com/RootstockCollective/collective-rewards-sc/blob/main/src/gauge/Gauge.sol#L342)
- [coinbase](https://github.com/RootstockCollective/collective-rewards-sc/blob/main/src/gauge/Gauge.sol#L366)

Comment (nmagariang) [reply src/SponsorsManager.sol:240]: I believe that separation in the gauge came from the fact that we could incentivize with the wrong token/send coinbase not accounted for, so besides being more explicit regarding the options to claim, I think it's not really necessary here

**Review by nmagariang — APPROVED** [review]
LGTM!

**Review by antomor — APPROVED** [review]
LGTM

**Pull request merged**